src/main/java/cs505/grad1/sensoragg/SensorAggHashMap.java

```diff
@@ -17,12 +15,55 @@
  */
 public class SensorAggHashMap extends HashMap<Integer, SensorData> {
 
-    //WELL-BEHAVED METHODS
+    private DataIterator iterator;
+
+    /**
+     * getValue will iterate over each entry in the SensorAggHashMap to find an entry of the
+     * matching type. It will then return the value in that entry as a double.
+     *
+     * @param sensorType is the SensorType enum reflecting the type of sensor that is being viewed
+     * @return the value detected by the sensor, represented as a double
+     */
+    public double getValue(SensorType sensorType) {
+        SensorData data;
+        do
+        {
+            data = getIterator().next().getValue();
+            if (data.getSensorType() == sensorType)
+                return data.getValue();
+        }
+        while(getIterator().hasNext());
+
+        //USING JAVA ITERATOR
+       // for (Map.Entry<Integer, SensorData> entry : this.entrySet())
+       // {
+       //     SensorData data = entry.getValue();
+       //     if (data.getSensorType() == sensorType)
+       //         return data.getValue();
+       // }
+        //TODO: Throw sensor not included exception?
+        return -999;
+    }
+
+    /**
+     * getIterator will access this instance's iterator object, or create one if none exists
+     *
+     * @return the iterator for this instance, as a DataIterator
+     */
+    private DataIterator getIterator() {
```

Comment:
Return type should be java.util.Iterator bound to the type returned

Reply:
Is there a reason this couldn't just return sag.entrySet().iterator() bound to the appropriate class instead?